pkg/policies/branch/branch.go

```diff
@@ -568,6 +595,42 @@ func fix(ctx context.Context, rep repositories, c *github.Client,
 				Str("area", polName).
 				Msg("Updated with Fix action.")
 		}
+
+		sp, rsp, err := rep.GetSignaturesProtectedBranch(ctx, owner, repo, b)
+		if err != nil {
+			if rsp != nil && rsp.StatusCode == http.StatusForbidden {
+				log.Warn().
```

Comment:
Usually we either warn and continue, or return a non-nil error, not both.

Both of these seem like errors that we should pass up, there should be at least a minimal branch protection setup on this branch at this point. If we want to add info to the error, we can wrap it.

Reply:
> Both of these seem like errors that we should pass up

Do you mean "pass up" in the call stack by returning them or "pass up" as in skip them by warning and continuing?

Reply:
I was trying to mimic the pattern of error handling seen here, but that may have been naive

https://github.com/ossf/allstar/blob/6d2a249050ba0f2ef8453e7c73037f360728f652/pkg/policies/branch/branch.go#L583-L594

Reply:
I assumed that you meant we should optionally wrap the errors and return them. But happy to fix if I have misunderstood.

Reply:
Yep, that looks good for the `RequireSignaturesOnProtectedBranch` as you have now.